src/kernel/src/common/start.c

```diff
@@ -39,32 +44,51 @@ void init() {
     asm volatile("pop {lr}");
 }
 
+
 /// Entrypoint for the C part of the kernel.
 /// This function is called by the assembly located in [startup.s].
 /// The MMU has already been initialized here but only the first MiB of the kernel has been mapped.
-void start(uint32_t * p_bootargs, size_t memory_size) {
+void start(uint32_t * p_bootargs, struct DTHeader * dtb) {
```

Comment:
maybe not do this in start.c (including the global variable). Might be better to move that to a seperate file which which exports a function `get_DTB()` or similar. `start.c` is nicer if it's jut a large list of function calls to set up the system, and preferably has as little control flow as possible.

Reply:
I've moved most of this to hardware info. However, do you mean that I should not pass the dtb header pointer to start? How else could I do it? That pointer must go from the bootloader to startup.s to start.c